src/julia.h

```diff
@@ -397,6 +397,7 @@ typedef struct {
     unsigned exportp:1;
     unsigned imported:1;
     unsigned deprecated:1;
+    jl_sym_t *deprecation_preferred_name;  // non-NULL when new type incompatible with old
```

Comment:
I looked at this comment more carefully:

> I think this can just replace `b->deprecated`

It can't. They aren't redundant; `deprecation_preferred_name` is almost always `NULL`. Only time it's not is when you want to engage in the kind of chicanery in the OP.